pkg/controller/history/controller_history.go

```diff
@@ -244,20 +247,25 @@ func (rh *realHistory) CreateControllerRevision(parent metav1.Object, revision *
 	}
 	clone := any.(*apps.ControllerRevision)
 
+	// First attempt to create the revision will not use collisionCount
```

Comment:
This seems different than how DaemonSet appears to use collisionCount, as far as I can see:

1. DaemonSet uses the collisionCount on the first attempt, if it was already set in Status.
2. DaemonSet saves the collisionCount to Status immediately upon incrementing it, instead of waiting until the retry loop eventually terminates.

Are these differences intentional? Have you thought about which way is better? Is this code meant to eventually be used generically by other controllers like DS too?